pkg/controller/cronjob/cronjob_controllerv2.go

```diff
@@ -467,13 +469,6 @@ func (jm *ControllerV2) syncCronJob(
 		// the job is missing in the lister but found in api-server
 	}
 
-	updatedCJ, err := jm.cronJobControl.UpdateStatus(ctx, cj)
```

Comment:
A few questions that we should be answered before we can proceed with this change:
- What was the rational to have the partial updates in the first place?
- What is each existing update trying to snapshot?
- Are we losing information by removing the partial updates in the case of an error?

Some of these questions should be answered in the PR description.

Reply:
Q: What was the rational to have the partial updates in the first place?
A: I think the rational is that, it's a clear and simple implementation way. I mean udpate cj object to api server as soon as we change it, it's easy for implementation, the logic is clear, and it is a much less error-prone way, especailly for exception case, as we do not need consider if partial updates is needed prior to exit.
But, it's not an efficient way, we should be sending ONE(or zero) update request per cronjob per cycle. bacause api-server qps is a scarce resource that we should optimize its use to be able to scale.

Reply:
Q: What is each existing update trying to snapshot?

A: The existing update locates in three places, two in syncCronJob() and another in cleanupFinishedJobs().

> In syncCronJob(), the first updates is trying to: 

-  snapshot Status.Active(the current running jobs list) of cronjob object after removing the missing job

- snapshot Status.LastSuccessfulTime of cronjob object in case we need process the timestamp for the finished job.

> In syncCronJob(), the second updates is trying to: 

- snapshot Status.Active(the current running jobs list) of cronjob object after deleting job that was still running at next scheduled start time in case of ReplaceConcurrent policy

- snapshot Status.Active(the current running jobs list) of cronjob object after creating a new job 



> In cleanupFinishedJobs(), the existing updates is trying to:

-  snapshot Status.Active(the current running jobs list) of cronjob object, after deleting the history redundant successful and failed job, and removing the reference in the active list

Reply:
Q: Are we losing information by removing the partial updates in the case of an error?

A: Yes, it's a bug, we are losing information in the case of an error. I have fixed it by adding updates in error processing.

Reply:
Answers of question 1 and 2 have been added into PR description.